[bug]: Dockerfile does not build; dependency conflict between angular-eslint/builder@12.2.1 and angular/cli@12.2.17

**Describe the bug**
A clear and concise description of what the bug is.

**To Reproduce**
Steps to reproduce the behavior:
1. Clone RTL repo
2. `cp dockerfiles/Dockerfile .`
3. `docker build .`
4. See error

```
 => ERROR [builder 5/9] RUN npm install                                                                                                                                                                2.1s
------                                                                                                                                                                                                      
 > [builder 5/9] RUN npm install:                                                                                                                                                                           
#10 1.967 npm ERR! code ERESOLVE
#10 1.969 npm ERR! ERESOLVE could not resolve
#10 1.970 npm ERR! 
#10 1.970 npm ERR! While resolving: @angular-eslint/builder@12.2.1
#10 1.970 npm ERR! Found: @angular/cli@13.1.1
#10 1.970 npm ERR! node_modules/@angular/cli
#10 1.970 npm ERR!   dev @angular/cli@"^13.0.3" from the root project
#10 1.970 npm ERR!   peer @angular/cli@">= 13.0.0 < 14.0.0" from @angular-eslint/schematics@13.0.1
#10 1.970 npm ERR!   node_modules/@angular-eslint/schematics
#10 1.970 npm ERR!     dev @angular-eslint/schematics@"13.0.1" from the root project
#10 1.970 npm ERR! 
#10 1.970 npm ERR! Could not resolve dependency:
#10 1.970 npm ERR! peer @angular/cli@">= 12.0.0 < 13.0.0" from @angular-eslint/builder@12.2.1
#10 1.970 npm ERR! node_modules/@angular-eslint/builder
#10 1.970 npm ERR!   dev @angular-eslint/builder@"^12.2.1" from the root project
#10 1.973 npm ERR! 
#10 1.973 npm ERR! Conflicting peer dependency: @angular/cli@12.2.17
#10 1.973 npm ERR! node_modules/@angular/cli
#10 1.973 npm ERR!   peer @angular/cli@">= 12.0.0 < 13.0.0" from @angular-eslint/builder@12.2.1
#10 1.973 npm ERR!   node_modules/@angular-eslint/builder
#10 1.973 npm ERR!     dev @angular-eslint/builder@"^12.2.1" from the root project
#10 1.973 npm ERR! 
#10 1.973 npm ERR! Fix the upstream dependency conflict, or retry
#10 1.973 npm ERR! this command with --force, or --legacy-peer-deps
#10 1.973 npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
#10 1.973 npm ERR! 
#10 1.973 npm ERR! See /root/.npm/eresolve-report.txt for a full report.
#10 1.973 
#10 1.974 npm ERR! A complete log of this run can be found in:
#10 1.974 npm ERR!     /root/.npm/_logs/2022-07-05T20_11_58_130Z-debug-0.log
------
executor failed running [/bin/sh -c npm install]: exit code: 1
```

**Your environment**
* Version of `RTL`: f84719e2c30312239e6a65fbc4662d29617dc2db
* Operating system (`uname -a` on *Nix) MacOS, building within linux docker container (also happens on my linux box)
* any other relevant environment details

**Additional context**
This build has succeeded every time for the last two years using our Dockerfile here: https://github.com/Start9Labs/ride-the-lightning-wrapper/blame/master/Dockerfile

Oddly this build succeeds when I switch back to using the node 10 base image rather than the node 16 one. (We switched from node 10 to 16 two months ago). It also succeeds if I add the suggested `--force` flag to the build, but I can't imagine either of these resolutions is optimal.

This build was succeeding up until just a few days ago, I'm guessing some recent upstream dependency update has caused this.

--- a/Dockerfile
+++ b/Dockerfile
@@ -9,7 +9,7 @@ COPY ./RTL/package.json /RTL/package.json
 COPY ./RTL/package-lock.json /RTL/package-lock.json
 
 # Install dependencies
-RUN npm install --only=prod
+RUN npm install --only=prod --legacy-peer-deps
 RUN wget https://github.com/mikefarah/yq/releases/download/v4.25.1/yq_linux_arm.tar.gz -O - |\
   tar xz && mv yq_linux_arm /usr/bin/yq
 

